### PR TITLE
Add pkg layering and admin unlock to sanity test

### PR DESCRIPTION
**Pull request by mike-nguyen**
Adding the package layering and ostree admin unlock tests to the
improved sanity tests.  The roles that were created for package
layering and admin unlock were leverage for basic sanity tests
through upgrades and rollbacks.

There was an issue with package layering a package from the EPEL
repo and the etc_modify role where the EPEL repo was added into
/etc.  This caused upgrades to fail because it could not find the
GPG KEY for the EPEL repo.  The EPEL repo GPG key is now copied
in as a part of the etc_modify role to work around this issue.

One more additional change was adding retries to the rpm-ostree
upgrade because flaky responses from Fedora repo caused the test
to intermittently fail with Socket I/O Timeout.

### CODE REVIEW

**Comment by miabbott** [comment]
[(3) [CI] [a-h-t] Add admin unlock and package layering sanity test to improved sanity tests](https://trello.com/c/TCEwEWlr/621-3-ci-a-h-t-add-admin-unlock-and-package-layering-sanity-test-to-improved-sanity-tests)

File: tests/improved-sanity-test/main.yml [hunk 1]
```diff
@@ -185,6 +216,23 @@
...
+      package: "{{ g_pkg }}"
+      tags:
+        - rpm_ostree_install_verify
+
+    # Check admin unlock overlayfs is no longer there after upgrade
+    - role: overlayfs_verify_missing
+      tags:
+        - overlayfs_verify_missing
+
+    # Verify admin unlocked package still installed
```
Comment (miabbott) [inline tests/improved-sanity-test/main.yml]: Seems like it should read:

`# Verify admin unlocked package is removed after upgrade`

**Comment by miabbott** [comment]
One minor nit, but otherwise this LGTM

**Comment by cgwalters** [comment]
Seems sane to me.

**Comment by mike-nguyen** [comment]
@miabott updated the comment

**Comment by jlebon** [comment]
Nice!

**Comment by miabbott** [comment]
Ran this on CAHC, F25AH, and RHELAH.  Ship it!

**Comment by mike-nguyen** [comment]
Thanks for looking at it!

**Pull request merged**